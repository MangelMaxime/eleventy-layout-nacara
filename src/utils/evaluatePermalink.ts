
export const evaluatePermalink = (ctx: any) => {
    let permalink = null;
    if (ctx.data.permalink) {
        if (typeof ctx.data.permalink === "function") {
            permalink = ctx.data.permalink(ctx.data)
        } else {
            permalink = ctx.data.permalink;
        }
    }
    return permalink;
}
