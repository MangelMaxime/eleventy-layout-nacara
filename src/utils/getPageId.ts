import path from "path";
// import NacaraTypes from "./../../types/nacara";

export default function getPageId(fileStem : string) {
    //  Normal the path, so we can split using the path separator
    const normalizedInputPath = path.normalize(fileStem);
    // Extract all the segments of the path
    const inputPathSegments = normalizedInputPath.split(path.sep);
    // console.log("inputPathSegments:", inputPathSegments);
    // Build the section direction, which consist of the root + the first segment of the path
    const pageIdSegments = inputPathSegments.slice(2);

    return pageIdSegments.join("/");
};
