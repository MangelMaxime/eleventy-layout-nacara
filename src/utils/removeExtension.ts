import path from "path";

export function removeExtension(fileName: string): string {
    const parsedPath = path.parse(fileName);
    return path.join(parsedPath.dir, parsedPath.name);
}
