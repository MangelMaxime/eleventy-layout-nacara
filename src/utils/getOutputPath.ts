import { evaluatePermalink } from "./evaluatePermalink.js";

export default function getOutputPath(pageContext: any) {
    const trimOutputRegex = new RegExp(`^${pageContext.data.eleventy.directories.output}`);
    const permalink = evaluatePermalink(pageContext);

    if (permalink) {
        return permalink;
    }

    return pageContext.data.page.outputPath.replace(trimOutputRegex, "");
}
