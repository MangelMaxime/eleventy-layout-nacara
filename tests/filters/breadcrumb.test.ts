import { expect, test } from 'vitest'
const Eleventy = require("@11ty/eleventy");
import { formatHTML } from "../utils/_formatHTML";

test("returns nothing if there is no menu", async () => {
    const elev = new Eleventy(
        "./fixtures/breadcrumb-0/",
        "./fixtures/breadcrumb-0/_site",
        {
            configPath: "./fixtures/breadcrumb-0/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const formattedResult = formatHTML(json[0].content);

    expect(formattedResult).toMatchSnapshot();
});

test("works for one level deep elements", async () => {
    const elev = new Eleventy(
        "./fixtures/breadcrumb-1/",
        "./fixtures/breadcrumb-1/_site",
        {
            configPath: "./fixtures/breadcrumb-1/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const formattedResult = formatHTML(json[0].content);

    expect(formattedResult).toMatchSnapshot();
});

test("works for multiple level deep elements", async () => {
    const elev = new Eleventy(
        "./fixtures/breadcrumb-2/",
        "./fixtures/breadcrumb-2/_site",
        {
            configPath: "./fixtures/breadcrumb-2/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const formattedResult = formatHTML(json[0].content);

    expect(formattedResult).toMatchSnapshot();
});
