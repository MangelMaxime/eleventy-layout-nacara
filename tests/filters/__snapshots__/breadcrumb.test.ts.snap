// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`returns nothing if there is no menu 1`] = `""`;

exports[`works for multiple level deep elements 1`] = `
"<nav class="breadcrumb">
    <ul>
        <li class="is-active"><a>Getting started</a></li>
        <li class="is-active"><a>Part 1</a></li>
        <li class="is-active"><a>Part 2</a></li>
        <li class="is-active"><a>Page</a></li>
    </ul>
</nav>
"
`;

exports[`works for one level deep elements 1`] = `
"<nav class="breadcrumb">
    <ul>
        <li class="is-active"><a>Getting started</a></li>
        <li class="is-active"><a>Page</a></li>
    </ul>
</nav>
"
`;
